### DISCUSSION TITLE
cycleway width quest appears on "virtual" link pieces not tagged as such

### DISCUSSION THREAD

**Discussion by joshinils** [streetcomplete/StreetComplete#4953] [Q&A]
See: https://www.openstreetmap.org/note/3645747#c7863707

The problem is that for this piece: https://osm.org/way/910784136
there was no `cycleway=link` tag present when the user Duedi was asked the width-quest, I added that afterward.

Is the width-quest disabled for pieces with `footway|path|cycleway=link|crossing`?

I think it would be a good idea to add an option to tag these link-pieces with the corresponding `*=link|crossing` tag, and thus disable the width-quest.

I think this would also benefit the tagging of intersections, since many have cycleways in an intersection where there is no defined width to be measured. and the `cycleway=link|crossing` would be a good fit there as an additional tag anyway.

**Comment by joshinils** [comment]
it does look like `cycleway=link` disables the width-quest, I like that.
I would then like to see an option to add that for pieces for which one can not answer the width-quest

**Reply by mnalis** [reply]
Yes, it is ignored on `*=link` ways, see: https://github.com/streetcomplete/StreetComplete/blob/1320cab649d434e9eee03f8db1ee12f851a2b936/app/src/main/java/de/westnordost/streetcomplete/quests/width/AddCyclewayWidth.kt#L35

There is always `Other Answers` / `Can't say` / `Leave note` to make a note that it is a link, and later edit at home with more advanced editor.  

I'm not sure how popular would be idea  to offer in StreetComplete to tag it directly from StreetComplete, as that would require explaining (in at maximum of three to four words!) what that answer entails (note that the average StreetComplete target user usually does not need to know _anything_ about OSM, Tags or highway infrastructure lingo). 
[SCEE fork](https://github.com/Helium314/SCEE) is geared more towards OSM experts, so that suggestion there would likely have much higher chances of being accepted there.

**Reply by joshinils** [reply]
> I'm not sure how popular would be idea to offer in StreetComplete to tag it directly from StreetComplete, as that would require explaining (in at maximum of three to four words!) what that answer entails (note that the average StreetComplete target user usually does not need to know _anything_ about OSM, Tags or highway infrastructure lingo).

I thought that might be a problem.

As is, the quest may confuse those non-OSM people for those non-link-tagged link-pieces. Leaving notes on them is of course an option, as has happened here.

-----

Overall, the existence of the link-pieces is not great, but there seems to be no better solution. Even for routing purposes, it seems the best right now.
Connecting the cycleway to the opposite kerb and jsut letting it end there is wrong, connecting it only until the crossing road is wrong and may lead to routing instructions like turn left onto road, and turn right onto other road.

**Reply by westnordost** [reply]
```
"It's still on the street"
  1   2     3    4     5
```
?
